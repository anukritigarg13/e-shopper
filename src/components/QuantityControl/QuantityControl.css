.quantity-control{
    display: flex;
    justify-content: space-around;
    background-color: aquamarine
}
.control-button{
    height: 2rem;
    width: 2rem;
}