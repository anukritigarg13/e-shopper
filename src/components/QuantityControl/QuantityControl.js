import React from 'react';
import PropTypes from 'prop-types';
import './QuantityControl.css';

const QuantityControl = ({ add, remove, quantity }) => (
  <div className="quantity-control" data-testid="quantity-control">
    <p><button className="control-button" onClick={add} type="button">+</button></p>
    <p className="quantity-in-basket">
      {quantity}
      {' '}
      in Basket
    </p>
    <p><button className="control-button" onClick={remove} type="button">-</button></p>

  </div>
);

QuantityControl.propTypes = {
  quantity: PropTypes.number.isRequired,
  add: PropTypes.func.isRequired,
  remove: PropTypes.func.isRequired,
};

export default QuantityControl;
