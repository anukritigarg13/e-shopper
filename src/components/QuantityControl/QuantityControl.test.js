import {
  screen, render, fireEvent,
} from '@testing-library/react';
import React from 'react';
import QuantityControl from './QuantityControl';

describe('QuantityControl component', () => {
  let MOCK_ADD; let MOCK_QUANTITY; let
    MOCK_REMOVE;
  beforeEach(() => {
    MOCK_QUANTITY = 5;
    MOCK_ADD = jest.fn();
    MOCK_REMOVE = jest.fn();
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should display count of items in basket', () => {
    render(<QuantityControl
      quantity={MOCK_QUANTITY}
      add={MOCK_ADD}
      remove={MOCK_REMOVE}
    />);
    screen.getByText('5 in Basket');
  });
  it('should display an add button', () => {
    render(<QuantityControl
      quantity={MOCK_QUANTITY}
      add={MOCK_ADD}
      remove={MOCK_REMOVE}
    />);
    const addElement = screen.getByText('+');
    expect(addElement.tagName).toBe('BUTTON');
  });
  it('should display a remove button', () => {
    render(<QuantityControl
      quantity={MOCK_QUANTITY}
      add={MOCK_ADD}
      remove={MOCK_REMOVE}
    />);
    const addElement = screen.getByText('-');
    expect(addElement.tagName).toBe('BUTTON');
  });
  it('should call add function on clicking + button', () => {
    render(<QuantityControl
      quantity={MOCK_QUANTITY}
      add={MOCK_ADD}
      remove={MOCK_REMOVE}
    />);
    const addElement = screen.getByText('+');
    fireEvent.click(addElement);
    expect(MOCK_ADD).toHaveBeenCalledTimes(1);
  });
  it('should call remove function on clicking - button', () => {
    render(<QuantityControl
      quantity={MOCK_QUANTITY}
      add={MOCK_ADD}
      remove={MOCK_REMOVE}
    />);
    const removeElement = screen.getByText('-');
    fireEvent.click(removeElement);
    expect(MOCK_REMOVE).toHaveBeenCalledTimes(1);
  });
  it('should match snapshot', () => {
    const { container } = render(<QuantityControl
      quantity={MOCK_QUANTITY}
      add={MOCK_ADD}
      remove={MOCK_REMOVE}
    />);
    expect(container).toMatchSnapshot();
  });
});
