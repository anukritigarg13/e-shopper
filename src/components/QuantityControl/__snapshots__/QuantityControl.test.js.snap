// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityControl component should match snapshot 1`] = `
<div>
  <div
    class="quantity-control"
    data-testid="quantity-control"
  >
    <p>
      <button
        class="control-button"
        type="button"
      >
        +
      </button>
    </p>
    <p
      class="quantity-in-basket"
    >
      5
       
      in Basket
    </p>
    <p>
      <button
        class="control-button"
        type="button"
      >
        -
      </button>
    </p>
  </div>
</div>
`;
